src/exercise/04.js

```diff
@@ -65,18 +30,92 @@ function Board() {
         {renderSquare(7)}
         {renderSquare(8)}
       </div>
-      <button className="restart" onClick={restart}>
-        restart
-      </button>
     </div>
   )
 }
 
+function Info({ status, moves, currentStep, setCurrentStep }) {
+  function goToMove(step) {
+    // Note to self:
+    // When currentStep changes, all the derived states for squares/winner/etc will
+    // also change accordingly, which is why the history buttons change the board.
+    setCurrentStep(step);
+  }
+  return (
+    <div>
+      <div className="status">{status}</div>
+      <ol>
+        {moves.map((move, step) => {
+          let buttonText = step ? `Go to move #${step}` : `Go to start`;
+          let isCurrentStep = currentStep === step;
+          return (
+            <li key={move}>
+              <button
+                disabled={isCurrentStep}
+                onClick={() => goToMove(step)}
```

Comment:
`goToMove` just calls `setCurrentStep` so you can simplify this to just:
```suggestion
                onClick={() => setCurrentStep(step)}
```